plugin/metricstore/prometheus/metricstore/reader.go

```diff
@@ -60,16 +61,32 @@ type (
 		metricDesc        string
 		buildPromQuery    func(p promQueryParams) string
 	}
+
+	promClient struct {
+		api.Client
+		extraParams map[string]string
+	}
 )
 
-// NewMetricsReader returns a new MetricsReader.
-func NewMetricsReader(cfg config.Configuration, logger *zap.Logger, tracer trace.TracerProvider) (*MetricsReader, error) {
-	logger.Info("Creating metrics reader", zap.Any("configuration", cfg))
+// URL decorator enables adding additional query parameters to the request sent to prometheus backend
+func (p promClient) URL(ep string, args map[string]string) *url.URL {
+	u := p.Client.URL(ep, args)
+
+	query := u.Query()
+	for k, v := range p.extraParams {
+		query.Set(k, v)
```

Comment:
why Set and not Add? We are not asked to replace params, only append them.

Reply:
Makes sense. Changed it to Add and updated test accordingly.